lib/utils.js

```diff
@@ -97,6 +97,8 @@ var Utils = module.exports = {
     return lodash.cloneDeep(obj, function (elem) {
       // Unfortunately, lodash.cloneDeep doesn't preserve Buffer.isBuffer, which we have to rely on for binary data
       if (Buffer.isBuffer(elem)) { return elem; }
+      // Preserve data-types like `fn` across clones
+      if (elem instanceof Utils.fn) { return new Utils.fn(elem.fn, elem.args); }
```

Comment:
We only need to clone if the internal state can somehow be mutated later on. I don't believe this is the case for `fn`.
In any case, if we do go this route, it would be better to add a clone method() to the `fn` prototype and call that from here.